cli/script/command-executor.ts

```diff
@@ -1020,6 +1056,49 @@ function getPackageMetricsString(packageObject: PackageWithMetrics): string {
     return returnString;
 }
 
+function getReactNativeProjectAppVersion(platform: string, projectName: string): Promise<string> {
+    if (platform === "ios") {
+        try {
+            var infoPlistContents: string = fs.readFileSync(path.join("iOS", projectName, "Info.plist")).toString();
+        } catch (err) {
+            try {
+                infoPlistContents = fs.readFileSync(path.join("iOS", "Info.plist")).toString();
+            } catch (err) {
+                throw new Error(`Unable to find or read "Info.plist" in the "iOS/${projectName}" or "iOS" folders.`);
+            }
+        }
+        
+        var parsedInfoPlist: any = plist.parse(infoPlistContents);
+        if (parsedInfoPlist && parsedInfoPlist.CFBundleShortVersionString) {
+             if (semver.valid(parsedInfoPlist.CFBundleShortVersionString) === null) {
+                throw new Error("Please update \"Info.plist\" to use a semver-compliant \"CFBundleShortVersionString\", for example \"1.0.3\".");
+            } else {
+                return Q(parsedInfoPlist.CFBundleShortVersionString);
+            }
+        } else {
+            throw new Error("Unable to parse the \"CFBundleShortVersionString\" from \"Info.plist\".");
+        }
+    } else {
+        var buildGradlePath: string = path.join("android", "app", "build.gradle");
+        if (fileDoesNotExistOrIsDirectory(buildGradlePath)) {
+            throw new Error("Unable to find or read \"build.gradle\" in the \"android/app\" folder.");
+        }
+        
+        return g2js.parseFile(buildGradlePath)
+            .catch((err: Error) => {
+                throw new Error("Unable to parse the \"versionName\" from \"build.gradle\".");
+            })
+            .then((buildGradle: any) => {
+                var appVersion: string = buildGradle.android.defaultConfig.versionName.replace(/"/g, "").trim();
```

Comment:
This might throw undefined error if any of the properties are missing `android` `defaultConfig` `versionName`, may be add an exist check for those?